pkg/installation/utils.go

```diff
@@ -150,17 +150,58 @@ func (i *Installation) setAdminPassword() error {
 		return nil
 	}
 	encPass := base64.StdEncoding.EncodeToString([]byte(i.Options.Password))
-	_, err := i.k8s.Static().CoreV1().ConfigMaps("kyma-installer").Patch("installation-config-overrides", types.JSONPatchType,
-		[]byte(fmt.Sprintf("[{\"op\": \"replace\", \"path\": \"/data/global.adminPassword\", \"value\": \"%s\"}]", encPass)))
+
+	err := i.patchOverride("installation-config-overrides", "global.adminPassword", encPass)
+	if err != nil {
+		err = errors.Wrap(err, "Error setting admin password")
+	}
+	return err
+}
+
+func (i *Installation) setDomain() error {
+	if i.Options.Domain == "" || (i.Options.IsLocal && i.Options.Domain == defaultDomain) {
+		return nil
+	}
+	err := i.patchOverride("installation-config-overrides", "global.domainName", i.Options.Domain)
+	if err != nil {
+		err = errors.Wrap(err, "Error setting custom domain")
+		return err
+	}
+	err = i.patchOverride("cluster-certificate-overrides", "global.tlsCrt", i.Options.TLSCert)
+	if err != nil {
+		err = errors.Wrap(err, "Error setting custom TLS certificate")
+		return err
+	}
+	err = i.patchOverride("cluster-certificate-overrides", "global.tlsKey", i.Options.TLSKey)
+	if err != nil {
+		err = errors.Wrap(err, "Error setting custom TLS private key")
+	}
+	return err
+}
+
+func (i *Installation) setMinikubeIP() error {
+	if !i.Options.IsLocal {
+		return nil
+	}
+	err := i.patchOverride("installation-config-overrides", "global.minikubeIP", i.Options.LocalCluster.IP)
+	if err != nil {
+		err = errors.Wrap(err, "Error setting minikube IP")
+	}
+	return err
+}
+
+func (i *Installation) patchOverride(override string, key string, value string) error {
+	_, err := i.k8s.Static().CoreV1().ConfigMaps("kyma-installer").Patch(override, types.JSONPatchType,
+		[]byte(fmt.Sprintf("[{\"op\": \"replace\", \"path\": \"/data/%s\", \"value\": \"%s\"}]", key, value)))
 	if err != nil {
 		if apiErrors.IsNotFound(err) {
 			_, err = i.k8s.Static().CoreV1().ConfigMaps("kyma-installer").Create(&corev1.ConfigMap{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:   "installation-config-overrides",
+					Name:   override,
 					Labels: map[string]string{"installer": "overrides"},
 				},
 				Data: map[string]string{
-					"global.adminPassword": encPass,
+					key: value,
 				},
 			})
 		}
```

Comment:
Please make the error message on the next line generic as well.

Reply:
done